data_diff/diff_tables.py

```diff
@@ -99,38 +96,46 @@ def segment_by_checkpoints(self, checkpoints: List[DbKey]) -> List["TableSegment
 
         return tables
 
-        ## Calculate checksums in one go, to prevent repetitive individual calls
-        # selects = [t._make_select(columns=[Checksum(self._relevant_columns)]) for t in tables]
-        # res = self.database.query(Select(columns=selects), list)
-        # checksums ,= res
-        # assert len(checksums) == len(checkpoints) + 1
-        # return [t.new(_checksum=checksum) for t, checksum in safezip(tables, checksums)]
-
-    def new(self, _count=None, _checksum=None, **kwargs) -> "TableSegment":
+    def new(self, **kwargs) -> "TableSegment":
         """Using new() creates a copy of the instance using 'replace()', and makes sure the cache is reset"""
-        return self.replace(_count=None, _checksum=None, **kwargs)
-
-    @property
-    def count(self) -> int:
-        if self._count is None:
-            self._count = self.database.query(self._make_select(columns=[Count()]), int)
-        return self._count
+        return self.replace(**kwargs)
 
     @property
     def _relevant_columns(self) -> List[str]:
-        return (
-            [self.key_column]
-            + ([self.update_column] if self.update_column is not None else [])
-            + list(self.extra_columns)
+        extras = set(self.extra_columns)
+        if self.update_column:
+            extras.add(self.update_column)
+
+        return [self.key_column] + list(sorted(extras))
+
+    def count(self) -> Tuple[int, int]:
+        return self.database.query(self._make_select(columns=[Count()]), int)
+
+    def count_and_checksum(self) -> Tuple[int, int]:
+        start = time.time()
+        count, checksum = self.database.query(
+            self._make_select(columns=[Count(), Checksum(self._relevant_columns)]), tuple
         )
+        duration = time.time() - start
+        if duration > RECOMMENDED_CHECKSUM_DURATION:
```

Comment:
Hm... so at first I know I suggested 1s, but now I'm thinking that on e.g. Snowflake it's actually kind of OK and even expected for something to run 5-10s, but on MySQL/Postgres 5-10s is really bad... Yeah, I don't feel too strongly, just feel good that we have _something_ in place.